validator/attester/service.go

```diff
@@ -22,49 +26,78 @@ type Attester struct {
 	cancel         context.CancelFunc
 	assigner       assignmentAnnouncer
 	assignmentChan chan bool
+	p2p            shared.P2P
+	blockBuf       chan p2p.Message
 }
 
 // Config options for an attester service.
 type Config struct {
-	AssignmentBuf int
-	Assigner      assignmentAnnouncer
+	AssignmentBuf   int
+	Assigner        assignmentAnnouncer
+	BlockBufferSize int
 }
 
 // NewAttester creates a new attester instance.
-func NewAttester(ctx context.Context, cfg *Config) *Attester {
+func NewAttester(ctx context.Context, cfg *Config, validatorP2P shared.P2P) *Attester {
 	ctx, cancel := context.WithCancel(ctx)
 	return &Attester{
 		ctx:            ctx,
 		cancel:         cancel,
 		assigner:       cfg.Assigner,
 		assignmentChan: make(chan bool, cfg.AssignmentBuf),
+		p2p:            validatorP2P,
+		blockBuf:       make(chan p2p.Message, cfg.BlockBufferSize),
 	}
 }
 
 // Start the main routine for an attester.
-func (p *Attester) Start() {
+func (a *Attester) Start() {
 	log.Info("Starting service")
-	go p.run(p.ctx.Done())
+	go a.run(a.ctx.Done())
 }
 
 // Stop the main loop.
-func (p *Attester) Stop() error {
-	defer p.cancel()
+func (a *Attester) Stop() error {
+	defer a.cancel()
 	log.Info("Stopping service")
 	return nil
 }
 
+// CreateAttestation will cause the attester to attest to the provided beacon block.
+func (a *Attester) CreateAttestation(block *pbp2p.BeaconBlock) *pbp2p.AttestationRecord {
+	attestation := &pbp2p.AttestationRecord{}
```

Comment:
Does it make sense to use the same `AttestationRecord` data structure here? Here's the fields:

```
type AttestationRecord struct {
	Slot                 uint64 
	ShardId              uint64  
	JustifiedSlot        uint64   
	JustifiedBlockHash   []byte  
	ShardBlockHash       []byte  
	AttesterBitfield     []byte  
	ObliqueParentHashes  [][]byte 
}
```

This is a single attestation, so `AttesterBitfield` doesn't make sense. `ObliqueParentHashes` is important when the proposer includes an attestation for a different fork, but that's obviously unimportant here. Maybe we should create a different data structure for individual attestations.